### PR TITLE
create resource package, extract BatchCreateGenericResources

### PR DESCRIPTION
**Pull request by mrparkers**
I want to try to eventually move all resource related code into this package.

### CODE REVIEW

File: pkg/resource/generate.go [hunk 1]
```diff
@@ -0,0 +1,3 @@
+package resource
+
+//go:generate counterfeiter github.com/rode/rode/pkg/resource.Manager
```
Comment (mrparkers) [inline pkg/resource/generate.go:3]: this needed to be in a new package to avoid cyclic dependency imports

File: Makefile [hunk 2]
```diff
@@ -30,3 +30,4 @@ mocks:
 	mockgen -package mocks github.com/rode/grafeas-elasticsearch/go/v1beta1/storage/filtering Filterer > mocks/grafeasElasticsearch.go
 	go install github.com/maxbrunsfeld/counterfeiter/v6@v6.4.1
 	COUNTERFEITER_NO_GENERATE_WARNING="true" go generate mocks/generate.go
```
Comment (alexashley) [inline Makefile:32]:
```suggestion
	COUNTERFEITER_NO_GENERATE_WARNING="true" go generate ./...
```

File: main.go [hunk 3]
```diff
@@ -89,7 +91,10 @@ func main() {
 		logger.Fatal("failed to create Elasticsearch client", zap.Error(err))
 	}
 
-	rodeServer, err := server.NewRodeServer(logger.Named("rode"), grafeasClientCommon, grafeasClientProjects, opaClient, esClient, filtering.NewFilterer(), c.Elasticsearch)
+	esutilClient := esutil.NewClient(logger.Named("ESClient"), esClient)
```
Comment (mrparkers) [inline main.go:94]: used this as an opportunity to pull in the new es client package as well

File: mocks/generate.go [hunk 4]
```diff
@@ -16,3 +16,4 @@ package mocks
 
 //go:generate counterfeiter -o grafeas_client.go github.com/rode/rode/protodeps/grafeas/proto/v1beta1/grafeas_go_proto.GrafeasV1Beta1Client
 //go:generate counterfeiter -o grafeas_projects_client.go github.com/rode/rode/protodeps/grafeas/proto/v1beta1/project_go_proto.ProjectsClient
+//go:generate counterfeiter --fake-name FakeEsClient -o es_client.go github.com/rode/grafeas-elasticsearch/go/v1beta1/storage/esutil.Client
```
Comment (alexashley) [inline mocks/generate.go:19]: What are your thoughts on reusing the fake client from grafeas-elasticsearch (or more generally as we reuse code in different repositories)?

Then as we make changes to the client interface, we don't have to regenerate the fake and don't have to keep all of the versions of the fake client in Git history.

Comment (mrparkers) [reply mocks/generate.go:19]: I didn't even consider that, good idea.

**Review by alexashley — APPROVED** [review]

**Pull request merged**